### PR TITLE
Update dino_layers.py

### PR DESCRIPTION
**Pull request by keyakiluo**
Fix some minor errors in the docstring of DinoTransformerDecoder.

### CODE REVIEW

File: mmdet/models/layers/transformer/dino_layers.py [hunk 1]
```diff
@@ -54,9 +54,19 @@ def forward(self, query: Tensor, value: Tensor, key_padding_mask: Tensor,
                 regression results.
 
         Returns:
-            Tensor: Output queries of Transformer encoder, which is also
-            called 'encoder output embeddings' or 'memory', has shape
+            Tensor: Output queries of Transformer decoder, which is also
+            called 'decoder output embeddings' or 'memory', has shape
```
Comment (Li-Qingyun) [inline mmdet/models/layers/transformer/dino_layers.py:58]: I think `which is also called 'decoder output embeddings' or 'memory'` can be deleted. Note that `memory` is encoder outputs.

File: mmdet/models/layers/transformer/dino_layers.py [hunk 2]
```diff
@@ -54,9 +54,19 @@ def forward(self, query: Tensor, value: Tensor, key_padding_mask: Tensor,
                 regression results.
 
         Returns:
-            Tensor: Output queries of Transformer encoder, which is also
-            called 'encoder output embeddings' or 'memory', has shape
+            Tensor: Output queries of Transformer decoder, which is also
```
Comment (Li-Qingyun) [inline mmdet/models/layers/transformer/dino_layers.py:57]: Return type seems to be `Tuple[Tensor, Tensor]`, not `Tensor`.
Please refer to other docstrings such as those in `DeformableDetrTransnformerDecoder`.

Besides, `Output queries` --> `Output queries and references`

**Review by Li-Qingyun — COMMENTED** [review]
Nice! Thanks for your contribution!

**Comment by OpenMMLab-Assistant005** [comment]
Hi @keyakiluo ！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request closed without merging**